docs/setup.md

```diff
@@ -0,0 +1,702 @@
+0.  Click “Create” to create the schedule.
+
+Now apply the schedule to your instance’s persistent disk:
+
+4.  Go to the [GCE Disks page](
+    https://console.cloud.google.com/compute/disks).
+0.  Click on the name of the persistent disk for your instance.  (By
+    default it will have the same name you gave to your instance.)
+0.  Click the pencil icon to edit the disk.
+0.  Under Snapshot schedule, select the schedule you created in steps
+    1–3.
+0.  Click Save.
+
+## Set Up Machine and Install Code City
+
+These instructions assume you are using a GCE instance running Debian
+GNU/Linux 9 (stretch), but feel free to adapt to your particular
+set-up.
+
+1.  Log into your instance.  (See instructions in first section if
+    using GCE.)
+0.  If your machine has less than 2GB of memory (check with `free -h`;
+    the very first number shown is total RAM), you will need to create
+    a swap file (this is mandatory on `f1-micro` instances):
+    ```
+    sudo -i
+    fallocate -l 2G /swapfile
+    chmod 600 /swapfile
+    mkswap /swapfile
+    swapon /swapfile
+    swapon -s
+    sh -c 'echo "/swapfile none swap sw 0 0" >> /etc/fstab'
+    exit
+    ```
+0.  Check for and install system updates, then install
+    [nginx](https://en.wikipedia.org/wiki/Nginx) and
+    [git](https://git-scm.com/):
+    ```
+    sudo apt-get update
+    sudo apt-get upgrade –y
+    sudo apt-get install -y nginx git
+    ```
+0.  Optionally install a text editor of your choice.  Debian comes
+    with [`vim`](https://www.vim.org/) and
+    [`nano`](https://www.nano-editor.org/) preinstalled; Emacs users
+    might feel more at home with the lightweight editors
+    [`mg`](https://github.com/hboetes/mg),
+    [`jove`](https://github.com/jonmacs/jove) or
+    [`zile`](https://www.gnu.org/software/zile/), or opt for
+    `emacs-nox` which is GNU Emacs without X Windows bindings.
+    ```
+    sudo apt-get install mg
+    ```
+0.  Install [node.js](https://nodejs.org/).  Code City depends on
+    version 12, which is more recent than the version included in
+    Debian 10, so we will obtain it via [NodeSource](
+    https://github.com/nodesource/distributions):
+    ```
+    sudo -i
+    curl -sL https://deb.nodesource.com/setup_12.x | bash -
+    apt-get install -y nodejs
+    exit
+    ```
+0.  Verify the correct version of node is installed:
+    ```
+    $ node –-version v12.18.2
+    ```
+    (Actual version may be later than 12.18.)
+
+### Get TLS Certificates
+
+In order to allow incoming HTTPS connections, you will need an
+[TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) [server
+certificate](
+https://en.wikipedia.org/wiki/Public_key_certificate#TLS/SSL_server_certificate
+).  There are two types:
+
+*   An ordinary certificate covers one or more specific domain names,
+    like `www.example.org`.
+*   A wildcard certificate includes one or more wildcard domains, like
+    `*.example.org`.  If you’re using a wildcard DNS entry, a
```

Comment:
a...

Reply:
Actually, the whole "If you’re using a wildcard DNS entry, a" is surplus to requirements.